snooty/types.py

```diff
@@ -427,6 +440,29 @@ def merge_facets(
 
         return merged_facets
 
+    @staticmethod
+    def get_facet_display_name(
+        parent_facets: List[Facet], unparsed_facet: Dict[str, Any]
+    ) -> str:
+        try:
+            if unparsed_facet["category"] == "version":
+                return str(unparsed_facet["value"])
+            taxonomy_ref = asdict(taxonomy.TaxonomySpec.get_taxonomy())
+            for parent in parent_facets:
+                options_list = taxonomy_ref[parent.category] or []
+                taxonomy_ref = [x for x in options_list if x["name"] == parent.value][0]
+            options_list = taxonomy_ref[unparsed_facet["category"]]
+            taxonomy_ref = [
+                x for x in options_list if x["name"] == str(unparsed_facet["value"])
+            ][0]
+            return str(
+                taxonomy_ref["display_name"]
+                or str(taxonomy_ref["name"]).replace(r"\_", " ").title()
```

Comment:
Do we need to account for `\-` for examples like `data-federation`? Is there an example facet that would use `\_`?

Reply:
good catch! updated the regex